src/NodeCompiler/Exception/UnableToCompileNode.php

```diff
@@ -5,7 +5,56 @@
 namespace Roave\BetterReflection\NodeCompiler\Exception;
 
 use LogicException;
+use PhpParser\Node;
+use Roave\BetterReflection\NodeCompiler\CompilerContext;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use function get_class;
+use function sprintf;
 
 class UnableToCompileNode extends LogicException
 {
+    public static function forUnRecognizedExpressionInContext(Node\Expr $expression, CompilerContext $context) : self
```

Comment:
The other two methods are named  `becauseOf<thing>`, but this method doesn't follow that same pattern. 
How about changing the name of this method to follow the same pattern?

Reply:
Could do, but at the call-site it reads better this way, so I'd prefer better readability from a consumer perspective than from a consistency across multiple named constructors